compiler/record-hessian/src/HessianComputer.cpp

```diff
@@ -0,0 +1,217 @@
+}
+
+void HessianComputer::recordHessianForConv2D(const luci::CircleNode *node)
+{
+  assert(_input_tensor->shape().num_dims() == 4);
+  assert(_input_tensor->element_type() == luci_interpreter::DataType::FLOAT32);
+
+  const auto circle_conv2d = loco::must_cast<const luci::CircleConv2D *>(node);
+  const auto node_filter = loco::must_cast<luci::CircleConst *>((circle_conv2d)->filter());
+  assert(circle_conv2d->rank() >= 4);
+  assert(node_filter->dtype() == loco::DataType::FLOAT32);
+  assert(node_filter->rank() == 4);
+
+  uint32_t size_filter = node_filter->size<loco::DataType::FLOAT32>();
+  uint32_t size_in_ch =
+    node_filter->size<loco::DataType::FLOAT32>() / circle_conv2d->dim(3).value();
+
+  uint32_t input_n = _input_tensor->shape().dim(0);
+  uint32_t input_h = _input_tensor->shape().dim(1);
+  uint32_t input_w = _input_tensor->shape().dim(2);
+  uint32_t input_c = _input_tensor->shape().dim(3);
+
+  uint32_t stride_h = circle_conv2d->stride()->h();
+  uint32_t stride_w = circle_conv2d->stride()->w();
+  uint32_t dilation_h = circle_conv2d->dilation()->h();
+  uint32_t dilation_w = circle_conv2d->dilation()->w();
+
+  uint32_t kernel_oc = node_filter->dim(0).value();
+  uint32_t kernel_h = node_filter->dim(1).value();
+  uint32_t kernel_w = node_filter->dim(2).value();
+  uint32_t kernel_ic = node_filter->dim(3).value();
+
+  const auto data = _input_tensor->data<float>();
+  const auto num_elements = _input_tensor->shape().num_elements();
+  assert(data != 0);
+  assert(num_elements != 0);
+  std::vector<float> buf(data, data + num_elements);
+
+  unfold(buf, input_n, input_h, input_w, input_c, stride_h, stride_w, dilation_h, dilation_w,
+         kernel_oc, kernel_h, kernel_w, kernel_ic);
+  assert(size_in_ch != 0);
+  uint32_t length = buf.size() / size_in_ch;
+
+  std::vector<float> hessian(size_in_ch * size_in_ch, 0);
+  for (int i = 0; i < size_in_ch; ++i)
+  {
+    for (int j = 0; j < size_in_ch; ++j)
+    {
+      float sum = 0;
+      for (int k = 0; k < length; ++k)
+      {
+        sum += buf[i + k * size_in_ch] * buf[j + k * size_in_ch];
+      }
+      hessian[i * size_in_ch + j] = 2 * sum;
+    }
+  }
+
+  HessianVector &vector = _hessian_map[node];
+  vector.update(hessian);
+}
+
+void HessianComputer::recordHessian(const luci::CircleNode *node,
+                                    const luci_interpreter::Tensor *input_tensor)
+{
+  if (node == nullptr || input_tensor == nullptr)
+    throw std::invalid_argument("RecordHessian: node or input_tensor is null.");
+
+  if (input_tensor->element_type() != luci_interpreter::DataType::FLOAT32)
+    throw std::runtime_error("RecordHessian: Unsupported dtype: only FLOAT32 is supported.");
+
+  _input_tensor = input_tensor;
+
+  switch (node->opcode())
+  {
+    case luci::CircleOpcode::FULLY_CONNECTED:
+      recordHessianForFullyConnected(node);
+      break;
+    case luci::CircleOpcode::CONV_2D:
+      recordHessianForConv2D(node);
+      break;
+    default:
```

Comment:
Q) I'm curious to see in #14285 with `CircleTransposeConv` Op but not here.
Are these two PRs not related?